src/components/Graph.tsx

```diff
@@ -184,20 +186,40 @@ class Graph extends React.Component <Props, States> {
     };
 
     setToReplica(e: any) {
-        if (Math.abs(e.target.getStage().getPointerPosition().y - 100) < 10 && e.target.getStage().getPointerPosition().x < 700) {
-            this.setState({toReplica: 0});
+        if (e.target.getStage().getPointerPosition().x < 100) {return;}
+        if (Math.abs(e.target.getStage().getPointerPosition().y - 100) < 25 && e.target.getStage().getPointerPosition().x < 700) {
+            this.setState({toReplica: 0, toY:100});
         }
-        if (Math.abs(e.target.getStage().getPointerPosition().y - 200) < 10 && e.target.getStage().getPointerPosition().x < 700) {
-            this.setState({toReplica: 1});
+        if (Math.abs(e.target.getStage().getPointerPosition().y - 200) < 25 && e.target.getStage().getPointerPosition().x < 700) {
+            this.setState({toReplica: 1,toY:200});
         }
-        if (Math.abs(e.target.getStage().getPointerPosition().y - 300) < 10 && e.target.getStage().getPointerPosition().x < 700) {
-            this.setState({toReplica: 2});
+        if (Math.abs(e.target.getStage().getPointerPosition().y - 300) < 25 && e.target.getStage().getPointerPosition().x < 700) {
+            this.setState({toReplica: 2,toY:300});
         }
         setTimeout(() => {
             this.setState({fromReplica: 'no replica', toReplica: 'no replica'});
         }, 1000);
 
     }
+
+
+setfromReplica(e: any) {
+        if (e.target.getStage().getPointerPosition().x < 100) {return;}
+        if (Math.abs(e.target.getStage().getPointerPosition().y - 100) < 25 && e.target.getStage().getPointerPosition().x < 700) {
+            this.setState({fromReplica: 0, fromY:100});
+        }
+        if (Math.abs(e.target.getStage().getPointerPosition().y - 200) < 25 && e.target.getStage().getPointerPosition().x < 700) {
+            this.setState({fromReplica: 1,fromY:200});
+        }
+        if (Math.abs(e.target.getStage().getPointerPosition().y - 300) < 25 && e.target.getStage().getPointerPosition().x < 700) {
+            this.setState({fromReplica: 2,fromY:300});
+        }
+        setTimeout(() => {
+            this.setState({fromReplica: 'no replica', toReplica: 'no replica'});
+        }, 2000);
```

Comment:
Why is this timeout needed? Does this mean I cannot draw a line if it takes more than 2 seconds?

Reply:
You have right, the timer is not a good idea. I used it as an easy way to reinitialize the fromReplica and toReplica 2 sec after the mouse click event and also 2 sec after the mouseup event. If I don't reinitialize the drawing doesn't work. 
Anyway I'll find a better solution

Reply:
I mean if u reinitialize directly and not after 2 sec in the mouse up event, nothing will be rendered because the fromReplica and toReplica are 'no replica'.
May be I can reinitialize on mouse down before drawing and not after finishing.